crates/analyzer/src/namespace/types.rs

```diff
@@ -494,6 +497,14 @@ impl Type {
         false
     }
 
+    pub fn is_zero_size(&self, db: &dyn AnalyzerDb) -> bool {
+        match &self {
+            Type::Base(Base::Unit) => true,
+            Type::Contract(inner) => inner.fields(db).is_empty(),
+            Type::Struct(inner) => inner.fields(db).is_empty(),
+            _ => false,
```

Comment:
`contract/struct` can be zero-sized even if their fields are not empty.

Example.
```
struct Zero {
    x: ()
}
```

Also, other types can be zero-sized.

* `Tuple` becomes a zero-sized type if it has no elements or all elements are zero-sized type.
* `Array` becomes a zero-sized type if its length is 0 or its element type is a zero-sized type.

Reply:
I create salsa query `is_zero_sized` to handle it. I think about this should be `type_size` will more general.